crates/bevy_transform/src/hierarchy/child_builder.rs

```diff
@@ -300,6 +339,48 @@ impl<'w> BuildWorldChildren for WorldChildBuilder<'w> {
         self.current_entity = self.parent_entities.pop();
         self
     }
+
+    fn push_children(&mut self, children: &[Entity]) -> &mut Self {
+        let parent = self
+            .current_entity
+            .expect("Cannot add children without a parent. Try creating an entity first.");
+        self.parent_entities.push(parent);
```

Comment:
We aren't creating a "nested WorldChildBuilder context" here like we do in `with_children`, so theres no need to push the parent entity (and in fact, that will break things).